quickwit/quickwit-storage/src/storage.rs

```diff
@@ -103,6 +103,13 @@ pub trait Storage: fmt::Debug + Send + Sync + 'static {
     /// Downloads a slice of a file from the storage, and returns an in memory buffer
     async fn get_slice(&self, path: &Path, range: Range<usize>) -> StorageResult<OwnedBytes>;
 
+    /// Open a stream handle on the file from the storage
+    async fn get_slice_stream(
```

Comment:
I'm surprised you wanted to stream slices and not the entire object. Is this correct?

Reply:
The file source seeks to the location of the last checkpoint, that's why I needed a `get_stream` with an arbitrary starting point. I actually don't need the end, but it makes things simpler to have it similar to the `get_slice`. I'm open to removing the end!